resources/exchanges.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+#
+# Cookbook Name:: rabbitmq
+# Resource:: exchanges
+#
+# Copyright 2011, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+actions :declare, :delete
+default_action :declare
+
+attribute :exchange, :kind_of => String, :name_attribute => true
+attribute :type, :kind_of => String, :name_attribute => true
+attribute :durable, :kind_of => [TrueClass, FalseClass], :name_attribute => true
+attribute :auto_delete, :kind_of => [TrueClass, FalseClass], :name_attribute => true
+attribute :vhost, :kind_of => String, :name_attribute => true
```

Comment:
Pretty sure you only want one of these attributes marked as a `name_attribute` for the LWRP

Reply:
Exchange name makes most sense as the name attribute. Of course there cannot be 4 named attributes.